### PR TITLE
[apex] Summit-AST Apex module - Part 4 - statements

### PR DESCRIPTION
**Pull request by aaronhurst-google**
Build and translate nodes related to statements.

Issue ref https://github.com/pmd/pmd/issues/3766

Confirmed that failing tests are expected (pending complete translation). This is ready for review and merge.

### CODE REVIEW

File: pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTExpressionStatement.java [hunk 1]
```diff
@@ -4,42 +4,16 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
-import com.google.summit.ast.statement.ExpressionStatement;
+import com.google.summit.ast.expression.Expression;
 
-public class ASTExpressionStatement extends AbstractApexNode.Single<ExpressionStatement> {
+public class ASTExpressionStatement extends AbstractApexNode.Single<Expression> {
```
Comment (adangel) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTExpressionStatement.java:9]: Is this change correct? From ExpressionStatement to Expression?

File: pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTVariableDeclarationStatements.java [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
+
 import net.sourceforge.pmd.annotation.InternalApi;
 
-public class ASTVariableDeclarationStatements extends AbstractApexNode.Single<Node> {
+import com.google.summit.ast.declaration.VariableDeclaration;
+
+public class ASTVariableDeclarationStatements extends AbstractApexNode.Many<VariableDeclaration> {
 
     @Deprecated
     @InternalApi
```
Comment (adangel) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTVariableDeclarationStatements.java:16]: Please remove these annotations and make the constructor package private

**Review by adangel — APPROVED** [review]
Thanks!

I'll merge this now, you can address my comments in the next PR.

File: pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTVariableDeclarationStatements.java (hunk 2 above)
Comment (aaronhurst-google) [reply pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTVariableDeclarationStatements.java:16]: Removed.

I noted a task to make a pass through all of the classes.

**Pull request merged**